app/src/main/java/com/ustwo/boilerplate/base/BaseActivity.java

```diff
@@ -0,0 +1,32 @@
+package com.ustwo.boilerplate.base;
+
+import android.os.Bundle;
+import android.support.annotation.LayoutRes;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.support.v7.app.AppCompatActivity;
+
+public abstract class BaseActivity<V extends BaseView> extends AppCompatActivity {
+
+  @Override
+  protected void onCreate(@Nullable Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+    setContentView(getLayoutId());
+    getPresenter().onViewAttached(getPresenterView());
+  }
+
+  @Override
+  protected void onDestroy() {
+    getPresenter().onViewDetached();
```

Comment:
Things like animation and video playback do sound like something that should be handled by callbacks where you want to explicitly state in the presenter / view contract what the intention is.

The intention of onViewDetached in this case is a very final method and should be used as merely a cleanup. Take for example switching between apps to copy and paste login info from a password manager, putting these calls in `onPause` would cause a full destruction of the presenter and subscriptions which is a fairly intense thing in and off itself.

Caching can mitigate the problem however it's not always required / not allowed by the business rules so this gives the opportunity to keep screens fresh. There is always the option for a user of the repo to override this functionality if it doesn't fit the use case.

In the case of custom views I was going to use the [learnings from Square](https://medium.com/square-corner-blog/android-leak-pattern-subscriptions-in-views-18f0860aa74c#.p9v89os3i) and other places. The issues on other projects were solved by `onWindowVisibilityChanged` but there were other causes to the issue.